### ISSUE TITLE
Write preprocessor

### ISSUE THREAD

**Issue by Mause** [galaxy-team/jupiter#3]

**Comment by Mause** [comment]
I had a look at the bison and flex combo, and it certainly doesn't seem very straight forward.

What did you use for libdcpu-cc @r4d2 ?

**Comment by patflick** [comment]
Neither.  Hold on,  I will get on IRC

**Comment by Mause** [comment]
Was it custom or a FOSS project?

**Comment by Mause** [comment]
Right, so @ThatOtherPerson is seeing if he can write a generic lexer parser thingy.
Seeing as we are yet to see it perform, i am unsure if we should continue with bison + flex or wait for @ThatOtherPerson's

**Comment by patflick** [comment]
Wait for now :-)

**Comment by Mause** [comment]
Well, in the meantime, i am going to clear the crap out of [Mause/Mausembler](http://github.com/Mause/Mausembler) so i can get into the right mindset for a cpp assembler

**Comment by Mause** [comment]
Though it would probably be worth it to start writing some unit tests so we can be all awesome and do TDD :D

**Comment by Mause** [comment]
Bah

**Comment by cassiebeckley** [comment]
Humbug
